CODEOWNERS

```diff
@@ -0,0 +1,45 @@
+# In this project, we use CODEOWNERS to identify people who are likely to know
+# who should review a pull request.
+#
+# People listed in this file are committing to respond in a timely fashion to
+# PRs in the selected areas. However, that response doesn't have to be a full
+# code review; it could also take any of these forms:
+#
+# - "I intend to review this but I can't yet. Please leave me a message if I
+#   haven't responded by (a specific date in the near future)."
+#
+# - "I think (a specific other contributor) should review this." (Note that the
+#   best reviewer for a PR may not necessarily be listed in this file.)
+#
+# People must only be added to this file if they've agreed to provide one of
+# the above responses in a reasonable amount of time for every PR to which
+# they're assigned.
+#
+# We only ask for this commitment from people who are employed full-time to
+# work on this project. We gratefully welcome reviews from other contributors,
+# but we don't believe it's fair to ask volunteers to respond quickly.
+
+# TODO: use GitHub teams instead of individual usernames
+
+# If none of the later patterns match, assign to anyone:
+* @alexcrichton @cfallin @elliottt @fitzgen @jameysharp @pchickey
+
+# Some parts of the project require more specialized knowledge. In those areas
+# we designate smaller groups who are more likely to be aware of who's working
+# in specific areas.
+
+# Wasmtime
+/crates/   @alexcrichton @fitzgen @jameysharp
```

Comment:
I think its worthwhile to break down the /crates further:
I will volunteer for component-macro, component-util, environ, fiber, runtime, test-programs, wasi, wasi-common, wasmtime, wiggle, wit-bindgen

wasi-nn and wasi-threads are maintained by @abrown, I will be a secondary on these in case he is unavailable
winch is @saulecabrera

Reply:
I wanted to limit the number of people in this initial PR so we can start evaluating how well this works for the project as a whole. If Andrew or Saúl would like to make the time commitment described in this file, we can certainly add them later.

Also, keep in mind that I'm not expecting the people in this file to necessarily be the ideal reviewers for the areas covered. Instead they should be people who know who the right reviewer is and, more importantly, can quickly get a response out so a contributor isn't waiting a long time. So it isn't as important to be precise about who's handling which area. I think it's more important to keep the lists simple so it's easy to tell at a glance who might get handed what.

In particular, given that you've volunteered for a third of the directories in /crates, I think you have a good idea who to hand reviews off to for the rest. So if it's alright with you, I'll just add you to the rotation for the whole Wasmtime group and drop the WASI category.

Reply:
@jameysharp I'm good with the time commitment described in this document and to be added to this list whenever you feel it's convenient. Thanks!